### ISSUE TITLE
Why does DeepAR predictions use mean and not the mode for prediction?

### ISSUE THREAD

**Issue by JosePeeterson** [jdb78/pytorch-forecasting#1169]
- PyTorch-Forecasting version: 

- PyTorch version: 
torch                         1.12.1+cu113
torchaudio                    0.12.1+cu113
torchmetrics                  0.10.2
torchsummary                  1.5.1
torchtext                     0.13.1
torchvision                   0.13.1+cu113

- Python version: 
python-apt                    0.0.0
python-dateutil               2.8.2
python-louvain                0.16
python-slugify                6.1.2
python-utils                  3.3.3
pytorch-lightning             1.6.0
- Operating System: Google colab

### Expected behavior

I am using the negative log liklihood loss for prediction of demand of count time series. In the code I noticed that the negbin distribution is sampled 100 times after gettting the mu and sigma from the LSTM network. The mean of 100 samples is used as prediction. But then is it not appropriate to use the mode of the distribtion? Since we are learning mu and sigma that MAXIMIZES the observed value in the decoder during during. 

### Actual behavior

However, result was .... I think it has to do with ... because of ...

### Code to reproduce the problem

In base_model.py
```
    def predict(
        self,
        data: Union[DataLoader, pd.DataFrame, TimeSeriesDataSet],
        mode: Union[str, Tuple[str, str]] = "prediction",
        return_index: bool = False,
        return_decoder_lengths: bool = False,
        batch_size: int = 64,
        num_workers: int = 0,
        fast_dev_run: bool = False,
        show_progress_bar: bool = False,
        return_x: bool = False,
        mode_kwargs: Dict[str, Any] = None,
        **kwargs,
    ):
        """
        Run inference / prediction.

        Returns:
            output, x, index, decoder_lengths: some elements might not be present depending on what is configured
                to be returned
        """
        with torch.no_grad():
            for x, _ in dataloader:
.
.
.
.
.

                elif mode == "prediction":
                    out = self.to_prediction(out, **mode_kwargs)
                    # mask non-predictions



```

```
    def to_prediction(self, out: Dict[str, Any], use_metric: bool = True, **kwargs):
        """
        Convert output to prediction using the loss metric.

        Args:
            out (Dict[str, Any]): output of network where "prediction" has been
                transformed with :py:meth:`~transform_output`
            use_metric (bool): if to use metric to convert for conversion, if False,
                simply take the average over ``out["prediction"]``
            **kwargs: arguments to metric ``to_quantiles`` method

        Returns:
            torch.Tensor: predictions of shape batch_size x timesteps
        """
        if not use_metric:
            # if samples were already drawn directly take mean
            # todo: support classification
            if isinstance(self.loss, MultiLoss):
                out = [Metric.to_prediction(loss, out["prediction"][idx]) for idx, loss in enumerate(self.loss)]
            else:
                out = Metric.to_prediction(self.loss, out["prediction"])


```

In base_metrics.py

```
    def to_prediction(self, y_pred: torch.Tensor) -> torch.Tensor:
        """
        Convert network prediction into a point prediction.

        Args:
            y_pred: prediction output of network

        Returns:
            torch.Tensor: point prediction
        """
        if y_pred.ndim == 3:
            if self.quantiles is None:
                assert y_pred.size(-1) == 1, "Prediction should only have one extra dimension"
                y_pred = y_pred[..., 0]
            else:
                y_pred = y_pred.mean(-1)
        return y_pred
```

Paste the command(s) you ran and the output. Including a link to a colab notebook will speed up issue resolution.
If there was a crash, please include the traceback here.
The code used to initialize the TimeSeriesDataSet and model should be also included.
